File: sdk/nodejs/tests/provider/experimental/testdata/complex-imports/path1/deep/component.ts
import * as pulumi from "@pulumi/pulumi";

export interface DeepComponentArgs {
    level: number;
    name?: string;
}

export class DeepComponent extends pulumi.ComponentResource {
    public readonly path: pulumi.Output<string>;

    constructor(name: string, args: DeepComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("test:component:DeepComponent", name, {}, opts);

        this.path = pulumi.output(`level-${args.level}/${args.name || "default"}`);

        this.registerOutputs({
            path: this.path,
        });
    }
}

File: tests/integration/component_provider/python/component-in-component/provider-nested/component.py
from typing import TypedDict

import pulumi


class Args(TypedDict):
    str_input: pulumi.Input[str]


class NestedComponent(pulumi.ComponentResource):
    str_output: pulumi.Output[str]

    def __init__(self, name: str, args: Args, opts: pulumi.ResourceOptions):
        super().__init__("component:index:NestedComponent", name, {}, opts)
        self.str_output = pulumi.Output.from_input(args.get("str_input")).apply(
            lambda x: x.upper()
        )
        self.register_outputs(
            {
                "str_output": self.str_output,
            }
        )
